datafusion/optimizer/src/eliminate_nested_union.rs

```diff
@@ -55,6 +55,9 @@ impl OptimizerRule for EliminateNestedUnion {
         _config: &dyn OptimizerConfig,
     ) -> Result<Transformed<LogicalPlan>> {
         match plan {
+            LogicalPlan::Union(Union { mut inputs, .. }) if inputs.len() == 1 => Ok(
```

Comment:
Worth adding a minor comment here, considering its not exactly related to nested unions?

(Or could rename away from `EliminateNestedUnions` to something like `SimplifyUnions` to be all encompassing)